crates/astria-sequencer/src/accounts/ics20_transfer.rs

```diff
@@ -0,0 +1,334 @@
+    async fn chan_open_init_execute<S: StateWrite>(_: S, _: &MsgChannelOpenInit) {}
+
+    async fn chan_open_try_execute<S: StateWrite>(_: S, _: &MsgChannelOpenTry) {}
+
+    async fn chan_open_ack_execute<S: StateWrite>(_: S, _: &MsgChannelOpenAck) {}
+
+    async fn chan_open_confirm_execute<S: StateWrite>(_: S, _: &MsgChannelOpenConfirm) {}
+
+    async fn chan_close_confirm_execute<S: StateWrite>(_: S, _: &MsgChannelCloseConfirm) {}
+
+    async fn chan_close_init_execute<S: StateWrite>(_: S, _: &MsgChannelCloseInit) {}
+
+    async fn recv_packet_execute<S: StateWrite>(mut state: S, msg: &MsgRecvPacket) {
+        use penumbra_ibc::component::packet::WriteAcknowledgement as _;
+
+        let ack = match execute_ics20_transfer(
+            &mut state,
+            &msg.packet.data,
+            &msg.packet.port_on_a,
+            &msg.packet.chan_on_a,
+            &msg.packet.port_on_b,
+            &msg.packet.chan_on_b,
+            false,
+        )
+        .await
+        {
+            Ok(()) => TokenTransferAcknowledgement::success(),
+            Err(e) => TokenTransferAcknowledgement::Error(e.to_string()),
+        };
+
+        let ack_bytes: Vec<u8> = ack.into();
+
+        _ = state
+            .write_acknowledgement(&msg.packet, &ack_bytes)
+            .await
+            .map_err(|e| {
+                tracing::error!("failed to write acknowledgement: {}", e);
+            });
+    }
+
+    async fn timeout_packet_execute<S: StateWrite>(mut state: S, msg: &MsgTimeout) {
+        // we put source and dest as chain_a (the source) as we're refunding tokens,
+        // and the destination chain of the refund is the source.
+        _ = execute_ics20_transfer(
+            &mut state,
+            &msg.packet.data,
+            &msg.packet.port_on_a,
+            &msg.packet.chan_on_a,
+            &msg.packet.port_on_a,
+            &msg.packet.chan_on_a,
+            true,
+        )
+        .await
+        .map_err(|e| {
+            tracing::error!(
+                "failed to refund tokens during timeout_packet_execute: {}",
+                e
+            );
+        });
+    }
+
+    async fn acknowledge_packet_execute<S: StateWrite>(mut state: S, msg: &MsgAcknowledgement) {
+        let ack: TokenTransferAcknowledgement = serde_json::from_slice(
+            msg.acknowledgement.as_slice(),
+        )
+        .expect("valid acknowledgement, should have been checked in acknowledge_packet_check");
+        if ack.is_successful() {
+            return;
+        }
+
+        // we put source and dest as chain_a (the source) as we're refunding tokens,
+        // and the destination chain of the refund is the source.
+        _ = execute_ics20_transfer(
+            &mut state,
+            &msg.packet.data,
+            &msg.packet.port_on_a,
+            &msg.packet.chan_on_a,
+            &msg.packet.port_on_a,
+            &msg.packet.chan_on_a,
+            true,
+        )
+        .await
+        .map_err(|e| {
```

Comment:
Prefer `if let Err(e) = ` over `map_err`. In the future we will have https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect_err